qa/workunits/rbd/compare_mirror_images.sh

```diff
@@ -0,0 +1,158 @@
+             | jq 'last' \
+             | jq 'select(.name | contains("mirror.primary"))' \
+             | jq 'select(.namespace.state == "demoted")')
+  if [[ $RBD_DEVICE_TYPE == "nbd" ]]; then
+    demote_id=$(echo $demote | jq -r '.id')
+    bdev=$(sudo rbd --cluster ${CLUSTER1} device map -t ${RBD_DEVICE_TYPE} \
+             --snap-id ${demote_id} ${POOL}/${img})
+  elif [[ $RBD_DEVICE_TYPE == "krbd" ]]; then
+    demote_name=$(echo $demote | jq -r '.name')
+    bdev=$(sudo rbd --cluster ${CLUSTER1} device map -t ${RBD_DEVICE_TYPE} \
+             ${POOL}/${img}@${demote_name})
+  else
+     echo "Unknown RBD_DEVICE_TYPE: ${RBD_DEVICE_TYPE}"
+     return 1
+  fi
+  demote_md5=$(sudo dd if=${bdev} bs=4M | md5sum | awk '{print $1}')
+  sudo rbd --cluster ${CLUSTER1} device unmap -t ${RBD_DEVICE_TYPE} ${bdev}
+
+  wait_for_demote_snap ${CLUSTER2} ${POOL} ${img}
+
+  promote_image ${CLUSTER2} ${POOL} ${img}
+
+  # promote non-primary image and calculate hash of its latest mirror snapshot
+  local promote promote_id promote_name promote_md5
+
+  promote=$(rbd --cluster ${CLUSTER2} snap ls --all ${POOL}/${img} --format=json \
+              | jq 'last' \
+              | jq 'select(.name | contains("mirror.primary"))')
+  if [[ $RBD_DEVICE_TYPE == "nbd" ]]; then
+    promote_id=$(echo $promote | jq -r '.id')
+    bdev=$(sudo rbd --cluster ${CLUSTER2} device map -t ${RBD_DEVICE_TYPE} \
+             --snap-id ${promote_id} ${POOL}/${img})
+  elif [[ $RBD_DEVICE_TYPE == "krbd" ]]; then
+    promote_name=$(echo $promote | jq -r '.name')
+    bdev=$(sudo rbd --cluster ${CLUSTER2} device map -t ${RBD_DEVICE_TYPE} \
+             ${POOL}/${img}@${promote_name})
+  else
+     echo "Unknown RBD_DEVICE_TYPE: ${RBD_DEVICE_TYPE}"
+     return 1
+  fi
+  promote_md5=$(sudo dd if=${bdev} bs=4M | md5sum | awk '{print $1}')
+  sudo rbd --cluster ${CLUSTER2} device unmap -t ${RBD_DEVICE_TYPE} ${bdev}
+
+  if [ "${demote_md5}" != "${promote_md5}" ]; then
+          return 1
+  fi
+}
+
+setup
+
+start_mirrors ${CLUSTER1}
+start_mirrors ${CLUSTER2}
+
+for i in {1..10}; do
+  for j in {1..10}; do
+    IMG=${IMG_PREFIX}${j}
+    MNTPT=${MNTPT_PREFIX}${j}
+    create_image_and_enable_mirror ${CLUSTER1} ${POOL} ${IMG} \
+      ${RBD_MIRROR_MODE} 10G
+    BDEV=$(sudo rbd --cluster ${CLUSTER1} device map -t ${RBD_DEVICE_TYPE} \
+             ${POOL}/${IMG})
+    sudo mkfs.ext4 ${BDEV}
+    sudo mkdir -p ${MNTPT}
+    sudo mount ${BDEV} ${MNTPT}
+    # create mirror snapshots under I/O
+    launch_manual_msnaps ${CLUSTER1} ${POOL} ${IMG} &
+    run_bench ${MNTPT} ${WORKLOAD_TIMEOUT} &
+  done
+  wait
+
+  pids=''
+  for j in {1..10}; do
+    compare_images $j &
+    pids+=" $!"
+  done
+
+  for pid in $pids; do
+    wait "$pid"
+    RC=$?
+    echo $pid $RC
+    if [ $RC != 0 ]; then
+      exit $RC
+    fi
+  done
+
+  for j in {1..10}; do
+    IMG=${IMG_PREFIX}${j}
+    remove_image ${CLUSTER2} ${POOL} ${IMG}
```

Comment:
to ensure that the removal of the primary/promoted image propagates, maybe we need to wait for the `mirror image status` of the non-primary image to be in `up+replaying` state?
```
wait_for_status_in_pool_dir ${CLUSTER1} ${POOL} ${IMG} 'up+replaying'
// and then...
remove_image ${CLUSTER2} ${POOL} ${IMG}
```